Sources/App/Scenes/CarPlaySceneDelegate.swift

```diff
@@ -173,49 +86,16 @@ extension CarPlaySceneDelegate: CPTemplateApplicationSceneDelegate {
         from window: CPWindow
     ) {
         NotificationCenter.default.removeObserver(self)
-        Current.servers.remove(observer: self)
-    }
-}
-
-// MARK: - ServerObserver
-
-@available(iOS 16.0, *)
-extension CarPlaySceneDelegate: ServerObserver {
-    func serversDidChange(_ serverManager: ServerManager) {
-        defer {
-            updateServerListButton()
-        }
-
-        guard let server = getServer(id: serverId) else {
-            serverId = nil
-
-            if let server = getServer() {
-                setServer(server: server)
-            } else if interfaceController?.presentedTemplate != nil {
-                interfaceController?.dismissTemplate(animated: true, completion: nil)
-            } else {
-                showNoServerAlert()
-            }
-
-            return
-        }
-        setServer(server: server)
     }
 }
 
 @available(iOS 16.0, *)
 extension CarPlaySceneDelegate: CPInterfaceControllerDelegate {
     func templateWillDisappear(_ aTemplate: CPTemplate, animated: Bool) {
-        domainsListTemplate?.templateWillDisappear(template: aTemplate)
+        allTemplates.forEach { $0.templateWillDisappear(template: aTemplate) }
```

Comment:
should these check if the template matches the individual one before calling? it would help eliminate the 'if template == self.template' check in (most?) of the implementations

Reply:
I was implementing that when I remember some templates have child templates (like the domain list > entities list and the Areas list > entities list).
Those we don't have the reference in scene delegate so this "event" needs to get to them from their parents